tests/e2e/e2e_distribution_test.go

```diff
@@ -0,0 +1,88 @@
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (s *IntegrationTestSuite) testDistribution() {
+	chainEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+
+	validatorB := s.chainA.validators[1]
+	validatorBAddr := validatorB.keyInfo.GetAddress()
+
+	valOperAddressA := sdk.ValAddress(validatorBAddr).String()
+
+	delegatorAddress := s.chainA.genesisAccounts[2].keyInfo.GetAddress().String()
+
+	newWithdrawalAddress := s.chainA.genesisAccounts[3].keyInfo.GetAddress().String()
+	fees := sdk.NewCoin(uoraiDenom, sdk.NewInt(1000))
+
+	beforeBalance, err := getSpecificBalance(chainEndpoint, newWithdrawalAddress, uoraiDenom)
+	s.Require().NoError(err)
+	if beforeBalance.IsNil() {
+		beforeBalance = sdk.NewCoin(uoraiDenom, sdk.NewInt(0))
+	}
+
+	s.execSetWithdrawAddress(s.chainA, 0, fees.String(), delegatorAddress, newWithdrawalAddress, oraiHomePath)
+
+	// Verify
+	s.Require().Eventually(
+		func() bool {
+			res, err := queryDelegatorWithdrawalAddress(chainEndpoint, delegatorAddress)
+			s.Require().NoError(err)
+
+			return res.WithdrawAddress == newWithdrawalAddress
+		},
+		10*time.Second,
+		5*time.Second,
+	)
+
+	s.execWithdrawReward(s.chainA, 0, delegatorAddress, valOperAddressA, oraiHomePath)
+	s.Require().Eventually(
+		func() bool {
+			afterBalance, err := getSpecificBalance(chainEndpoint, newWithdrawalAddress, uoraiDenom)
+			s.Require().NoError(err)
+
+			return afterBalance.IsGTE(beforeBalance)
+		},
+		10*time.Second,
+		5*time.Second,
+	)
+}
+
+/*
+fundCommunityPool tests the funding of the community pool on behalf of the distribution module.
+Test Benchmarks:
+1. Validation that balance of the distribution module account before funding
+2. Execution funding the community pool
+3. Verification that correct funds have been deposited to distribution module account
+*/
+func (s *IntegrationTestSuite) fundCommunityPool() {
```

Comment:
why do we need basic e2e test cases that test cosmos sdk modules? I think we should only keep test cases related to the software upgrade & the ICS. Other test suites are not needed

Reply:
e2e test has been removed